OpenTelemetry (S3M) Web Servers apps (#1795)
SumoLogic/sumologic-documentation: docs/integrations/web-servers/opentelemetry/apache-tomcat-opentelemetry.md

---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/web-servers/apache-tomcat.png')} alt="Thumbnail icon" width="65"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

The [Apache Tomcat](https://tomcat.apache.org/tomcat-8.5-doc/index.html) app is a logs app that helps you get insight into visitor locations, traffic patterns, errors, resource utilization, garbage collection, web server operations and access from known malicious sources.

Tomcat logs are sent to Sumo Logic through Opentelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-Schematics.png')} alt="Schematics" />

## Fields Created in Sumo Logic for Tomcat

Following are the [Fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of Tomcat App install if not already present.

> - **`webengine.cluster`** - User configured. Set a value of the cluster where your tomcat instance resides. This will be tagged along with the data sent to sumo.

Reviewer: ```suggestion
- **`webengine.cluster.name`** - User configured. Set a value of the cluster where your tomcat instance resides. This will be tagged along with the data sent to sumo.
```

- **`webengine.system`** - Has a fixed value of **tomcat**.
- **`sumo.datasource`** - Has a fixed value of **tomcat**.

## Prerequisite

The Sumo Logic App for Apache Tomcat uses three types of logs

1.  Tomcat Access logs \ Log format description: [https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html \ ](https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html)Recommended pattern used is pattern="common"
2.  Tomcat Catalina.out logs \ Log format description: <https://docs.oracle.com/javase/8/docs/api/java/util/logging/SimpleFormatter.html>
3.  Tomcat Garbage Collection (GC) logs \ Log format description: <https://stackoverflow.com/questions/4468546/explanation-of-tomcat-gc-log-statements>

By default, Tomcat logs are stored in `/usr/share/tomcat/logs/` The default directory for log files is listed in the `/usr/share/tomcat/conf/logging.properties` file.

## Collecting Logs & installing Tomcat app

Here are the steps for Collecting Logs and installing the app:

### Step 1: Set up Collector

:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the "Use an existing Collector" option.
:::

> If you want to use an existing Otel Collector then this step can be skipped by selecting the option of using an existing Collector.

Reviewer: This line needs to be removed. Its duplicate of the previous line.

> If you want to create a new Collector please select **Add a new Collector** option.

Reviewer: ```suggestion
To create a new Collector:

1. Select the **Add a new Collector** option.
1. Select the platform for which you want to install the Sumo OpenTelemetry Collector.
```

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.  

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-Collector.png')} alt="Collector" />

### Step 2: Configure integration

In this step we will be configuring the yaml required for Tomcat Collection.

Path of the log file configured to capture tomcat logs is needed to be given here.

> The files are typically located in /usr/share/tomcat/logs/*. If you're using a customized path, check the Tomcat.conf file for this information.

Author: URL must be in code font

> 

Reviewer: ```suggestion
You can add any custom fields which you want to tag along with the data ingested in sumo.
```

Click on the **Download YAML File** button to get the yaml file.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-YAML.png')} alt="YAML" />

### Step 3: Sending logs and metrics to Sumo

Once you have the yaml file downloaded in step 2, you can copy it to the machine which needs to be monitored. Follow the below steps based on the platform of the machine:

<Tabs
  className="unique-tabs"
  defaultValue="Linux"
  values={[
    {label: 'Linux', value: 'Linux'},
    {label: 'Windows', value: 'Windows'},
    {label: 'macOS', value: 'macOS'},
  ]}>

<TabItem value="Linux">

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Tomcat instance which needs to be monitored.

> 2.  restart the collector using

Author: Capital R

```sh
sudo systemctl restart otelcol-sumo
```

</TabItem>
<TabItem value="Windows">

1.  Copy the yaml file to `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d` folder in the machine which needs to be monitored.
2.  Restart the collector using 
```sh
Restart-Service -Name OtelcolSumo
```

</TabItem>
<TabItem value="macOS">

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Tomcat instance which needs to be monitored.
2.  Restart the otelcol-sumo process using the below command 
```sh
 otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```

</TabItem>
</Tabs>

After successful execution of the above command, Sumo will start receiving the data from your host machine.

Click **Next**. This will install the app to your Sumo Logic Org. The app consists of Dashboards.

> Panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but within 20 minutes, you'll see full graphs and map

Author: Sentence needs to end with period

---
id: apache-tomcat-opentelemetry
title: Apache Tomcat - OpenTelemetry Collector
sidebar_label: Apache Tomcat - OTel Collector
description: Learn about the Sumo Logic OpenTelemetry App for Apache Tomcat.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/web-servers/apache-tomcat.png')} alt="Thumbnail icon" width="65"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

The [Apache Tomcat](https://tomcat.apache.org/tomcat-8.5-doc/index.html) app is a logs app that helps you get insight into visitor locations, traffic patterns, errors, resource utilization, garbage collection, web server operations and access from known malicious sources.

Tomcat logs are sent to Sumo Logic through Opentelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-Schematics.png')} alt="Schematics" />

Reviewer 2: This image link is broken.

## Fields Created in Sumo Logic for Tomcat

> Following are the [Fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of Tomcat App install if not already present.

Author: ```suggestion
The following are the [Fields](/docs/manage/fields) that will be created as part of the Tomcat App install, if not already present.
```

> - **`webengine.cluster.name`** - User configured. Set a value of the cluster where your tomcat instance resides. This will be tagged along with the data sent to sumo.

Reviewer 2: ```suggestion
- **`webengine.cluster.name`** - User configured. Set a value of the cluster where your Tomcat instance resides. This will be tagged along with the data sent to Sumo.
```

- **`webengine.system`** - Has a fixed value of **tomcat**.
- **`sumo.datasource`** - Has a fixed value of **tomcat**.

## Prerequisite

The Sumo Logic App for Apache Tomcat uses three types of logs

> 1.  Tomcat Access logs \ Log format description: [https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html \ ](https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html)Recommended pattern used is pattern="common"

Reviewer 2: ```suggestion
1.  Tomcat Access logs \ Log format description: [https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html](https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html) Recommended pattern used is pattern="common"
```

2.  Tomcat Catalina.out logs \ Log format description: <https://docs.oracle.com/javase/8/docs/api/java/util/logging/SimpleFormatter.html>
3.  Tomcat Garbage Collection (GC) logs \ Log format description: <https://stackoverflow.com/questions/4468546/explanation-of-tomcat-gc-log-statements>

By default, Tomcat logs are stored in `/usr/share/tomcat/logs/` The default directory for log files is listed in the `/usr/share/tomcat/conf/logging.properties` file.

## Collecting Logs and Installing the Apache Tomcat app

Here are the steps for Collecting Logs and installing the app:

### Step 1: Set up Collector

:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the "Use an existing Collector" option.
:::

> If you want to use an existing Otel Collector then this step can be skipped by selecting the option of using an existing Collector.

Author: ```suggestion
```

To create a new Collector:

1. Select the **Add a new Collector** option.
1. Select the platform for which you want to install the Sumo OpenTelemetry Collector.

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.  

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-Collector.png')} alt="Collector" />

Reviewer 2: This image link is broken.

### Step 2: Configure integration

In this step we will be configuring the yaml required for Tomcat Collection.

Path of the log file configured to capture tomcat logs is needed to be given here.

The files are typically located in /usr/share/tomcat/logs/*. If you're using a customized path, check the Tomcat.conf file for this information.
You can add any custom fields which you want to tag along with the data ingested in sumo.
Click on the **Download YAML File** button to get the yaml file.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Tomcat-OpenTelemetry/Apache-Tomcat-YAML.png')} alt="YAML" />

Reviewer 2: This image link is broken.

### Step 3: Sending logs and metrics to Sumo

Once you have the yaml file downloaded in step 2, you can copy it to the machine which needs to be monitored. Follow the below steps based on the platform of the machine:

<Tabs
  className="unique-tabs"
  defaultValue="Linux"
  values={[
    {label: 'Linux', value: 'Linux'},
    {label: 'Windows', value: 'Windows'},

>     {label: 'macOS', value: 'macOS'},

Reviewer 2: ```suggestion
    {label: 'Mac', value: 'Mac'},
```

  ]}>

<TabItem value="Linux">

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Tomcat instance which needs to be monitored.

> 2.  restart the collector using

Reviewer 2: ```suggestion
2.  Restart the collector using:
```

```sh
sudo systemctl restart otelcol-sumo
```

</TabItem>
<TabItem value="Windows">

1.  Copy the yaml file to `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d` folder in the machine which needs to be monitored.

> 2.  Restart the collector using 

Reviewer 2: ```suggestion
2.  Restart the collector using:
```

```sh
Restart-Service -Name OtelcolSumo
```

</TabItem>

> <TabItem value="macOS">

Reviewer 2: ```suggestion
<TabItem value="Mac">
```

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Tomcat instance which needs to be monitored.

> 2.  Restart the otelcol-sumo process using the below command 

Reviewer 2: ```suggestion
2.  Restart the otelcol-sumo process using the below command:
```

```sh
 otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```

</TabItem>
</Tabs>

After successful execution of the above command, Sumo will start receiving the data from your host machine.

Click **Next**. This will install the app to your Sumo Logic Org. The app consists of Dashboards.

> Panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but within 20 minutes, you'll see full graphs and map

Reviewer 2: ```suggestion
Panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but within 20 minutes, you'll see full graphs and maps.
```
